k9mail-library/src/main/java/com/fsck/k9/mail/transport/SmtpTransport.java

```diff
@@ -660,74 +664,19 @@ private void writeLine(String s, boolean sensitive) throws IOException {
         mOut.flush();
     }
 
-    private void checkLine(String line) throws MessagingException {
-        int length = line.length();
-        if (length < 1) {
-            throw new MessagingException("SMTP response is 0 length");
-        }
-
-        char c = line.charAt(0);
-        if ((c == '4') || (c == '5')) {
-            int replyCode = -1;
-            String message = line;
-            if (length >= 3) {
-                try {
-                    replyCode = Integer.parseInt(line.substring(0, 3));
-                } catch (NumberFormatException e) { /* ignore */ }
-
-                if (length > 4) {
-                    message = line.substring(4);
-                } else {
-                    message = "";
-                }
-            }
-
-            throw new NegativeSmtpReplyException(replyCode, message);
-        }
-
-    }
-    @Deprecated
-    private List<String> executeSimpleCommand(String command) throws IOException, MessagingException {
-        return executeSimpleCommand(command, false);
-    }
-
-    /**
-     * TODO:  All responses should be checked to confirm that they start with a valid
-     * reply code, and that the reply code is appropriate for the command being executed.
-     * That means it should either be a 2xx code (generally) or a 3xx code in special cases
-     * (e.g., DATA & AUTH LOGIN commands).  Reply codes should be made available as part of
-     * the returned object.
-     *
-     * This should be done using the non-deprecated API below.
-     */
-    @Deprecated
-    private List<String> executeSimpleCommand(String command, boolean sensitive)
-    throws IOException, MessagingException {
-        List<String> results = new ArrayList<>();
-        if (command != null) {
-            writeLine(command, sensitive);
-        }
-
-        String line = readCommandResponseLine(results);
-
-        // Check if the reply code indicates an error.
-        checkLine(line);
-
-        return results;
-    }
-
     private static class CommandResponse {
 
         private final int replyCode;
-        private final String message;
+        private final List<String> results;
 
-        public CommandResponse(int replyCode, String message) {
+        public CommandResponse(int replyCode, List<String> results) {
             this.replyCode = replyCode;
-            this.message = message;
+            this.results = results;
         }
     }
 
-    private CommandResponse executeSimpleCommandWithResponse(String command, boolean sensitive) throws IOException, MessagingException {
+    private CommandResponse executeSimpleCommand(String command, boolean sensitive)
```

Comment:
hm, I'm not too fond of the `sensitive` parameter here, not sure why the non parametrized version was marked as deprecated. it seems easy to mess this up, is more often than not false, and requires a lookup to find out what the parameter does. Maybe have `executeSimpleCommandSensitive` instead?

Related note, given that many of the parameters used as input are concatenated strings, it might also be nice to have string format style varargs here.